jac/jaclang/compiler/parser.py

```diff
@@ -680,48 +684,40 @@ def enum(self, kid: list[ast.AstNode]) -> ast.Enum | ast.EnumDef:
             enum: decorators? enum_decl
                 | enum_def
             """
-            if isinstance(kid[0], ast.SubNodeList):
-                if isinstance(kid[1], ast.Enum):
-                    kid[1].decorators = kid[0]
-                    kid[1].add_kids_left([kid[0]])
-                    return kid[1]
-                else:
-                    raise self.ice()
-            elif isinstance(kid[0], (ast.Enum, ast.EnumDef)):
-                return kid[0]
+            decorator = self.match(ast.SubNodeList)
+            enum_node: ast.Enum | ast.EnumDef | None = None
+            if decorator is not None:
+                enum_node = self.consume(ast.Enum)
+                enum_node.decorators = decorator
+                enum_node.add_kids_left([decorator])
+            else:
+                enum_node = self.match(ast.Enum) or self.match(ast.EnumDef)
+            if enum_node is not None:
+                return enum_node
             else:
-
                 raise self.ice()
 
         def enum_decl(self, kid: list[ast.AstNode]) -> ast.Enum:
             """Grammar rule.
 
             enum_decl: KW_ENUM access_tag? STRING? NAME inherited_archs? (enum_block | SEMI)
             """
-            access = kid[1] if isinstance(kid[1], ast.SubTag) else None
-            semstr = (
-                kid[2]
-                if (access and isinstance(kid[2], ast.String))
-                else kid[1] if isinstance(kid[1], ast.String) else None
-            )
-            name = (
-                kid[3]
-                if (access and semstr)
-                else kid[2] if (access or semstr) else kid[1]
+            self.node_idx = 1
```

Comment:
You shouldn't modify `node_idx`, treat it like private and call `self.consume_token(Tok.KW_ENUM)`, that'll internally increment the node index (which will be gone in the future). Also it ensure the first token is keyword "enum".